theama/optical_flow/farneback.py

```diff
@@ -36,13 +36,20 @@ def perform_optical_flow(self, video):
         Args:
             video: is the video fed in as a numpy array.
         """
+        if type(video) is not np.ndarray:
+            raise Exception("Not a numpy array")
+        if len(video.shape) < 3 or len(video.shape) > 4:
+            raise Exception("Not a video numpy file")
+
         video = video.astype('uint8')
         f, r, c, d = video.shape
```

Comment:
run all your code against pylint to put proper variable names. This is all open-source so we need some standards. Install pylint: pip install pylint. Then in the project folder run pylint theama and it will tell you issues.